src/snowflake/cli/app/telemetry.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import annotations
+
+import platform
+import sys
+from enum import Enum, unique
+from typing import Any, Dict, Union
+
+import click
+from snowflake.cli.__about__ import VERSION
+from snowflake.cli.api.cli_global_context import cli_context
+from snowflake.cli.api.output.formats import OutputFormat
+from snowflake.cli.api.utils.error_handling import safe
+from snowflake.connector.telemetry import (
+    TelemetryData,
+    TelemetryField,
+)
+from snowflake.connector.time_util import get_time_millis
+
+
+@unique
+class CLITelemetryField(Enum):
+    # Basic information
+    SOURCE = "source"
+    VERSION_CLI = "version_cli"
+    VERSION_PYTHON = "version_python"
+    VERSION_OS = "version_os"
+    # Command execution context
+    COMMAND = "command"
+    COMMAND_GROUP = "command_group"
+    COMMAND_FLAGS = "command_flags"
+    COMMAND_OUTPUT_TYPE = "command_output_type"
+    # Information
+    EVENT = "event"
+    ERROR_MSG = "error_msg"
+    ERROR_TYPE = "error_type"
+
+
+class TelemetryEvent(Enum):
+    CMD_EXECUTION = "executing_command"
```

Comment:
Just asking: is this event type ok without mentioning that the event is about "snowcli command"? I know that we have `source: snowcli` but I just wonder if event type should be somehow connected to SnowCLI.

Reply:
I took a look at the other implementations (snowpark and connecotr) and there's no strict relation.